xdcs-server/src/main/java/pl/edu/agh/xdcs/services/TaskService.java

```diff
@@ -86,12 +92,13 @@ public void reportFailure(String taskId) {
         historicalTaskDao.setErrored(taskId);
     }
 
-    public void saveLog(Task task, Instant time, LogLineEntity.LogType type, byte[] contents) {
+    public void saveLog(Task task, Instant time, LogLineEntity.LogType type, byte[] contents, Agent agent) {
         LogLineEntity logLine = new LogLineEntity();
         logLine.setTask(task.asHistorical());
         logLine.setType(type);
         logLine.setTime(time);
         logLine.setContents(contents);
+        logLine.setAgent(agentDao.getAgent(agent));
```

Comment:
to jako argument można pobrać AgentEntity zamiast robić query dla każdej linii logów..